Cython/Compiler/Nodes.py

```diff
@@ -4184,11 +4184,14 @@ def generate_tuple_and_keyword_parsing_code(self, args, code, decl_code):
         code.putln('{')
         all_args = tuple(positional_args) + tuple(kw_only_args)
         non_posonly_args = [arg for arg in all_args if not arg.pos_only]
-        non_pos_args_id = ','.join(
-            ['&%s' % code.intern_identifier(arg.entry.name) for arg in non_posonly_args] + ['0'])
-        code.putln("PyObject ** const %s[] = {%s};" % (
-            Naming.pykwdlist_cname,
-            non_pos_args_id))
+        if non_posonly_args:
+            non_pos_args_id = ','.join([
+                f'&{code.intern_identifier(arg.entry.name)}'
+                for arg in non_posonly_args
+            ] + ['0'])
+            code.putln("PyObject ** const %s[] = {%s};" % (
+                Naming.pykwdlist_cname,
+                non_pos_args_id))
```

Comment:
Here is a different proposal:
```suggestion
        if non_posonly_args:
            non_posonly_args_init = '{%s,0}' % ','.join([
                f'&{code.intern_identifier(arg.entry.name)}'
                for arg in non_posonly_args
            ])
            code.putln(
                f"PyObject ** const {Naming.pykwdlist_cname}[] = {non_posonly_args_init};")
```